com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/TimeoutAsyncFlowableStep.java

```diff
@@ -28,9 +28,19 @@ private boolean hasTimedOut(DelegateExecution context) {
     private long getStepStartTime(DelegateExecution context) {
         Long stepStartTime = (Long) context.getVariable(getStepStartTimeVariable());
         if (stepStartTime == null) {
-            stepStartTime = System.currentTimeMillis();
-            context.setVariable(getStepStartTimeVariable(), stepStartTime);
+            return retrieveAndSetStartTime(context);
         }
+
+        if (StepsUtil.getStepPhase(context) == StepPhase.RETRY) {
```

Comment:
Merge this if statement with the one above. Then, inline the retrieveAndSetStartTime() method in the body of the statement.

Reply:
Thought for this... the code is in this style due to more readability - will check how it will look like inlined